Common/Sources/View/Error/ErrorView.swift

```diff
@@ -17,21 +17,34 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 import HyperwalletSDK
-import UIKit
 
 /// The class to handle UI errors
 public final class ErrorView {
-    weak var viewController: UIViewController!
-    var error: HyperwalletErrorType
+    private let errorTypeConnection = "CONNECTION"
 
-    /// Initializer to initialize the class with errors to be displayed and the viewcontroller responsible
+    private weak var viewController: UIViewController!
+    private var error: HyperwalletErrorType
+    private var pageName: String
+    private var pageGroup: String
+    private let hyperwalletInsights: HyperwalletInsightsProtocol
+
+    /// Initializer to initialize the class with errors to be displayed and the ViewController responsible
     /// to display the errors
     /// - Parameters:
     ///   - viewController: view controller that contains errors
     ///   - error: hyperwallet error
-    public init(viewController: UIViewController, error: HyperwalletErrorType) {
+    ///   - pageName: The Page or screen that is currently visible
+    ///   - pageGroup: The group of the Page or screen that is currently visible
+    public init(viewController: UIViewController,
+                hyperwalletInsights: HyperwalletInsightsProtocol = HyperwalletInsights.shared,
```

Comment:
why do you need this in init? You can just use it like HyperwalletInsights.shared.trackError